R/datamodelr-code.R

```diff
@@ -428,26 +440,51 @@ to_html_table <- function(x,
 dot_html_label <- function(x, title, palette_id = "default", col_attr = c("column"),
                            columnArrows = FALSE) {
   cols <- c("ref", col_attr)
-  if (is.null(palette_id)) {
+  if (is.null(palette_id) || palette_id == "show") {
     palette_id <- "default"
   }
-
-  border <- ifelse(is.null(bdm_color(palette_id, "line_color")), 0, 1)
+  # currently we always have a border around the tables
+  border <- 1
+  # test if palette_id is valid: either datamodelr or hexcode (converted in `dm_set_colors()` from colorname)
+  if (palette_id == "default") {
+    col <- list(
+      line_color = "#555555",
+      header_bgcolor = "#EFEBDD",
+      header_font = "#000000",
+      bgcolor = "#FFFFFF"
+    )
+  } else {
+    header_bgcol_rgb <- col2rgb(palette_id)[, 1]
+    bodycol_rgb <- as.integer(header_bgcol_rgb + (255 - header_bgcol_rgb) * 0.87)
+    bodycol <- rgb(bodycol_rgb[1], bodycol_rgb[2], bodycol_rgb[3], maxColorValue = 255)
+    # if header background too dark, use white font color
+    header_font <- if (sum(header_bgcol_rgb) < 300) "#FFFFFF" else "#000000"
```

Comment:
```suggestion
    header_font <- if (is_dark_color(header_bcgol_rgb)) "#FFFFFF" else "#000000"
```

with a suitable `is_dark_color()` function.